### ISSUE TITLE
handle the PeerConnection.OnDataChannel event 

### ISSUE THREAD

**Issue by zhangazheng** [microsoft/MixedReality-WebRTC#78]
OnDataChannel event triggered when a remote peer opens a data channel,  any idea?

**Comment by djee-ms** [comment]
Sorry @zhangazheng I am not sure I understand the question. Do you mean that `OnDataChannel` is the event that is triggered when a new in-band negotiated data channel is received from the remote peer, but that there is no C# event corresponding to it in the `PeerConnection` C# object?

If that's it, yes indeed I just realized this should be added. This is already handled inside the C++ code, but there is no callback at the moment in C#. I will fix.

**Comment by zhangazheng** [comment]
Yes 👍,  I need the callback in C# .  So I can listen for the `OnDataChanne`l event to handle remote connection messages.

**Comment by djee-ms** [comment]
@zhangazheng this should be fixed with da153ea05f9b81d630899bd7db50f13dee136b1d, can you please try again and let me know if that works for you?

**Comment by zhangazheng** [comment]
@djee-ms  I will try it as soon as posible

**Comment by zhangazheng** [comment]
@djee-ms  👍 cool ,   it works fine, now I can receive message from remote data channel. 

But there are other problems that have occurred, first I create a data channel with `AddDataChannelAsync` ,  but Unity is crashed when I stop the unity editor player. 
Here is the Editor crash log:
```
========== OUTPUTTING STACK TRACE ==================

0x00007FFDDAB13BAD (Microsoft.MixedReality.WebRTC.N) [C:\w\libs\Microsoft.MixedReality.WebRTC.Native\src\peer_connection.cpp:255] Microsoft::MixedReality::WebRTC::PeerConnection::RemoveDataChannel 
0x00007FFDDAB11992 (Microsoft.MixedReality.WebRTC.N) Microsoft::MixedReality::WebRTC::DataChannel::`scalar deleting destructor'
0x00007FFDDAB17460 (Microsoft.MixedReality.WebRTC.N) [C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\xmemory:1033] std::_Destroy_range<std::allocator<std::shared_ptr<Microsoft::MixedReality::WebRTC::DataChannel> > > 
0x00007FFDDAB12A1F (Microsoft.MixedReality.WebRTC.N) [C:\w\libs\Microsoft.MixedReality.WebRTC.Native\src\peer_connection.cpp:130] Microsoft::MixedReality::WebRTC::PeerConnection::~PeerConnection 
0x00007FFDDAB16B39 (Microsoft.MixedReality.WebRTC.N) rtc::RefCountedObject<Microsoft::MixedReality::WebRTC::PeerConnection>::`scalar deleting destructor'
0x00007FFDDAB166A9 (Microsoft.MixedReality.WebRTC.N) [C:\w\external\webrtc-uwp-sdk\webrtc\xplatform\webrtc\rtc_base\refcountedobject.h:43] rtc::RefCountedObject<Microsoft::MixedReality::WebRTC::PeerConnection>::Release 
0x00007FFDDAB0F113 (Microsoft.MixedReality.WebRTC.N) [C:\w\libs\Microsoft.MixedReality.WebRTC.Native\src\api.cpp:1075] mrsPeerConnectionClose 
0x0000025F8CEAB427 (Mono JIT Code) (wrapper managed-to-native) Microsoft.MixedReality.WebRTC.Interop.PeerConnectionInterop:PeerConnection_Close (intptr&)
0x0000025F8CEAB0D3 (Mono JIT Code) [C:\w\libs\Microsoft.MixedReality.WebRTC\PeerConnection.cs:827] Microsoft.MixedReality.WebRTC.PeerConnection:Close () 
0x0000025F8CEAACF3 (Mono JIT Code) [C:\w\libs\Microsoft.MixedReality.WebRTC\PeerConnection.cs:847] Microsoft.MixedReality.WebRTC.PeerConnection:Dispose () 
0x0000025F8CEAA553 (Mono JIT Code) [D:\workspace\HololensStarLink\Assets\Microsoft.MixedReality.WebRTC.Unity\Scripts\PeerConnection.cs:304] Microsoft.MixedReality.WebRTC.Unity.PeerConnection:Uninitialize () 
0x0000025F8CEA9DEB (Mono JIT Code) [D:\workspace\HololensStarLink\Assets\Microsoft.MixedReality.WebRTC.Unity\Scripts\PeerConnection.cs:375] Microsoft.MixedReality.WebRTC.Unity.PeerConnection:OnDestroy () 
0x0000025F843B4098 (Mono JIT Code) (wrapper runtime-invoke) object:runtime_invoke_void__this__ (object,intptr,intptr,intptr)
0x00007FFDF7D7BBCB (mono-2.0-bdwgc) [c:\users\builduser\builds\_cjcc5tm\0\vm\mono\mono\mini\mini-runtime.c:2809] mono_jit_runtime_invoke 
0x00007FFDF7D02252 (mono-2.0-bdwgc) [c:\users\builduser\builds\_cjcc5tm\0\vm\mono\mono\metadata\object.c:2919] do_runtime_invoke 
0x00007FFDF7D0B25F (mono-2.0-bdwgc) [c:\users\builduser\builds\_cjcc5tm\0\vm\mono\mono\metadata\object.c:2966] mono_runtime_invoke 
0x00007FF6B319B106 (Unity) scripting_method_invoke
0x00007FF6B31953F5 (Unity) ScriptingInvocation::Invoke
0x00007FF6B319569E (Unity) ScriptingInvocation::InvokeChecked
0x00007FF6B31EF903 (Unity) SerializableManagedRef::CallMethod
0x00007FF6B3169906 (Unity) MonoBehaviour::WillDestroyComponent
0x00007FF6B2467E4E (Unity) GameObject::WillDestroyGameObject
0x00007FF6B2C7B535 (Unity) PreDestroyRecursive
0x00007FF6B2C78D0B (Unity) DestroyObjectHighLevel_Internal
0x00007FF6B2C78B2E (Unity) DestroyObjectHighLevel
0x00007FF6B287E8FB (Unity) DelayedDestroyCallback
0x00007FF6B2881CD8 (Unity) DelayedCallManager::Update
0x00007FF6B2CB0F2F (Unity) `InitPlayerLoopCallbacks'::`2'::UpdateScriptRunDelayedDynamicFrameRateRegistrator::Forward
0x00007FF6B2C9A3C8 (Unity) ExecutePlayerLoop
0x00007FF6B2C9A4A6 (Unity) ExecutePlayerLoop
0x00007FF6B2C9F060 (Unity) PlayerLoop
0x00007FF6B19BEBA0 (Unity) PlayerLoopController::UpdateScene
0x00007FF6B19BBC19 (Unity) Application::TickTimer
0x00007FF6B1C7A2EB (Unity) MainMessageLoop
0x00007FF6B1C83CF7 (Unity) WinMain
0x00007FF6B45C349E (Unity) __scrt_common_main_seh
0x00007FFE55D37BD4 (KERNEL32) BaseThreadInitThunk
0x00007FFE5678CEE1 (ntdll) RtlUserThreadStart

========== END OF STACKTRACE ===========
```

**Comment by djee-ms** [comment]
I fixed something similar in 42ad09eb864959399514e7306f1368c25cd9f728, and again in d32fd3e2c357cc49d1d2b8449223e34db10ad36b, do you have them? Sorry I didn't catch those until after I pushed the initial change.

**Comment by zhangazheng** [comment]
It's awesome, everything works fine.   @djee-ms  thank you very much.
